public/src/js/controllers/search.js

```diff
@@ -17,48 +17,56 @@ angular.module('insight.search').controller('SearchController',
     $scope.q = '';
     $scope.loading = false;
   };
+  
+  $scope.defineSearchType = function(searchString) {
+    if (searchString.length == 64) {
+      var block_flag = false;
+      for (var i = 0; i < 2; i++) {
+        if (searchString[i] == '0') {
```

Comment:
Not sure if this is an entirely great way of differentiating between blockhash and txid. How did this used to work previously? I'm almost certain this used to work but for whatever reason stopped working.

Reply:
Previously there was no check on the query string. It was just trying all API methods in a sequence until it works.

Reply:
Ok. Do you know why it stopped working? Something in the last release broke the behaviour, and I'd like to get to the bottom of whatever that was.

Reply:
The root cause of regression comes from a modification on https://github.com/bitpay/insight-api/pull/23 which disabled status 400 result on error. So the request is timing out. This will be fixed by https://github.com/dashevo/insight-api/tree/fix/unknown_res